grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy

```diff
@@ -599,6 +600,48 @@ class IndicatorDataService {
         return graphData;
     }
 
+    GraphData getPercentageAdHoc(Location location) {
+        Date today = new Date()
+        Integer previousMonth = today.month
+
+        // if previousMonth = december, we are in the previous year
+        Integer year = previousMonth == 11 ? today.year + 1899 : today.year + 1900
+
+        List<String> listLabels = []
+        List<Integer> listData = []
+        List<IndicatorDatasets> datasets = []
+       
+        def percentageAdHoc = Requisition.executeQuery("""
+            select count(r.id), r.type
+            from Requisition as r
+            where r.origin.id = :location
+            and MONTH(r.requestedDeliveryDate) = :previousMonth
+            and YEAR(r.requestedDeliveryDate) = :year
+            group by r.type
+        """,
+                [
+                        'location': location.id,
+                        'previousMonth'   : previousMonth,
+                        'year'    : year,
+                ]);
+
+        percentageAdHoc.each {
+            if (RequisitionType.listRequestTypes().contains(it[1])) {
+                listLabels.push(it[1].toString())
+                listData.push(it[0])
+            }     
+        }
+
+        IndicatorDatasets indicatorDatasets =  new IndicatorDatasets('Percentage', listData, null , 'doughnut')
+        datasets.push(indicatorDatasets)
+
+        IndicatorData indicatorData = new IndicatorData(datasets, listLabels)
+
+        GraphData graphData = new GraphData(indicatorData, 'Percentage Stock vs Ad Hoc Requests this Month', 'doughnut', '/openboxes/shipment/list')
```

Comment:
@julienreliefapps Can you change the link to 
```
/openboxes/stockMovement/list?direction=OUTBOUND
```

Reply:
@jmiranda  I just did it